modules/swagger-codegen/src/main/resources/rust/model.mustache

```diff
@@ -0,0 +1,57 @@
+{{>partial_header}}
+{{#models}}
+{{#model}}
+{{#description}}
+// {{{classname}}} : {{{description}}}
+{{/description}}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct {{classname}} {
+{{#vars}}
+  {{#description}}
+  // {{{description}}}
+  {{/description}}
+  #[serde(rename = "{{baseName}}")] {{name}}: {{^required}}Option<{{/required}}{{{datatype}}}{{^required}}>{{/required}}{{#hasMore}},{{/hasMore}}
+{{/vars}}
+}
+
+impl {{classname}} {
+  {{#description}}
+  // {{{description}}}
+  {{/description}}
+  pub fn new({{#requiredVars}}{{name}}: {{{datatype}}}{{^-last}}, {{/-last}}{{/requiredVars}}) -> {{classname}} {
+    {{classname}} {
+      {{#vars}}
+      {{name}}: {{#required}}{{name}}{{/required}}{{^required}}{{#isListContainer}}Vec:new(){{/isListContainer}}{{#isMapContainer}}Hash:new(){{/isMapContainer}}{{^isContainer}}None{{/isContainer}}{{/required}}{{#hasMore}},{{/hasMore}}
+      {{/vars}}
+    }
+  }
+
+  {{#vars}}
+  pub fn set_{{name}}(&mut self, {{name}}: {{{datatype}}}) {
+    self.{{name}} = Some({{name}});
```

Comment:
`Some(...)` only applies to `Option<>` (optional) fields. If the field isn't optional, assign as `self.{{name}} = {{name}}`